src/Microsoft.Data.SqlClient/netcore/src/Microsoft/Data/Common/DbConnectionOptions.cs

```diff
@@ -72,39 +57,33 @@ internal bool ConvertValueToIntegratedSecurityInternal(string stringValue)
             }
         }
 
-        public int ConvertValueToInt32(string keyName, int defaultValue)
+        public int ConvertValueToInt32(Dictionary<string, string> parsetable, string keyName, int defaultValue)
         {
-            string value;
-            return _parsetable.TryGetValue(keyName, out value) && value != null ?
-                ConvertToInt32Internal(keyName, value) :
-                defaultValue;
+	        string value;
+	        return parsetable.TryGetValue(keyName, out value) && value != null ?
+		        ConvertToInt32Internal(keyName, value) :
+		        defaultValue;
         }
 
         internal static int ConvertToInt32Internal(string keyname, string stringValue)
         {
-            try
-            {
-                return int.Parse(stringValue, System.Globalization.NumberStyles.Integer, CultureInfo.InvariantCulture);
-            }
-            catch (FormatException e)
-            {
-                throw ADP.InvalidConnectionOptionValue(keyname, e);
-            }
-            catch (OverflowException e)
-            {
-                throw ADP.InvalidConnectionOptionValue(keyname, e);
-            }
-        }
-
-        public string ConvertValueToString(string keyName, string defaultValue)
-        {
-            string value;
-            return _parsetable.TryGetValue(keyName, out value) && value != null ? value : defaultValue;
+	        try
+	        {
+		        return int.Parse(stringValue, System.Globalization.NumberStyles.Integer, CultureInfo.InvariantCulture);
+	        }
+	        catch (FormatException e)
+	        {
+		        throw ADP.InvalidConnectionOptionValue(keyname, e);
+	        }
+	        catch (OverflowException e)
+	        {
+		        throw ADP.InvalidConnectionOptionValue(keyname, e);
+	        }
         }
 
-        public bool ContainsKey(string keyword)
+        public string ConvertValueToString(Dictionary<string, string> parsetable, string keyName, string defaultValue)
```

Comment:
Can be made `internal static` too? Please review similar APIs as they come together as similar changes.

Reply:
done